front/src/applications/operationalStudies/hooks/useCachedTrackSections.ts

```diff
@@ -0,0 +1,39 @@
+import { useRef, useCallback } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { InfraObjectWithGeometry } from 'common/api/osrdEditoastApi';
+
+export default function useCachedTrackSections(infraId: number) {
+  const trackIdsRef = useRef<Set<string>>(new Set());
+  const trackSectionsRef = useRef<InfraObjectWithGeometry[]>([]);
```

Comment:
Do you think it would be smart to store only the railjson, so the type would be directly `TrackSection[]` ?